### PR TITLE
health check, dynamic reconnect, bugfix

### PR DESCRIPTION
**Pull request by tseglevskiy**

### CODE REVIEW

File: lib/stf/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module Stf
-  VERSION = '0.3.0.rc10'
+  VERSION = '0.3.0.rc12'
```
Comment (Malinskiy) [inline lib/stf/version.rb]: FYI: https://semver.org/spec/v2.0.0-rc.1.html

File: lib/stf/model/device_list.rb [hunk 2]
```diff
@@ -12,13 +12,26 @@ def initialize(devices)
...
     def byFilter(filter)
-      filter ? select {|d| d.checkFilter(filter)} : Array.new
+      filter ? select {|d| d.checkFilter(filter)} : []
     end
 
     def exceptFilter(filter)
       filter ? reject {|d| d.checkFilter(filter)} : this
     end
 
+    def healthy(pattern)
```
Comment (Malinskiy) [inline lib/stf/model/device_list.rb]: this should be named ```healthy?``` in Ruby

File: lib/stf/model/device_list.rb [hunk 3]
```diff
@@ -12,13 +12,26 @@ def initialize(devices)
...
+      pattern ? select { |d| d.isHealthy(pattern) } : this
+    end
+
+    # more pessimistic than healthy()
+    def healthyForConnect(pattern)
```
Comment (Malinskiy) [inline lib/stf/model/device_list.rb]: ```healthyForConnect?```

File: lib/stf/model/device_list.rb [hunk 4]
```diff
@@ -12,13 +12,26 @@ def initialize(devices)
...
+      pattern ? select { |d| d.isHealthyForConnect(pattern) } : this
+    end
+
+    def notHealthy(pattern)
```
Comment (Malinskiy) [inline lib/stf/model/device_list.rb]: why not just invert ```healthy```? Why do we need this?

File: lib/stf/model/device.rb [hunk 5]
```diff
@@ -8,6 +8,32 @@ def getKeys
       getKeysNextLevel('', self)
     end
 
+    # more pessimistic decision
+    def isHealthyForConnect(pattern)
```
Comment (Malinskiy) [inline lib/stf/model/device.rb]: ```healthyForConnect?```

File: lib/stf/model/device.rb [hunk 6]
```diff
@@ -8,6 +8,32 @@ def getKeys
...
+      health = isHealthy(pattern)
+      ppp = pattern.split(',')
+      ppp.each do |p|
+        health &&= getValue('battery.temp').to_i < 30 if ['t', 'temp', 'temperature'].include? p
+        health &&= getValue('battery.level').to_f > 30.0 if ['b', 'batt', 'battery'].include? p
+      end
+      health
+    end
+
+    def isHealthy(pattern)
```
Comment (Malinskiy) [inline lib/stf/model/device.rb]: ```healthy?```

**Review by Malinskiy — CHANGES_REQUESTED** [review]

File: lib/stf/version.rb (hunk 1 above)
Comment (tseglevskiy) [reply lib/stf/version.rb]: Thanks

File: lib/stf/model/device_list.rb (hunk 3 above)
Comment (tseglevskiy) [reply lib/stf/model/device_list.rb]: became `select_healthy_for_connect`

File: lib/stf/model/device.rb (hunk 5 above)
Comment (tseglevskiy) [reply lib/stf/model/device.rb]: became `healthy_for_connect?`

File: lib/stf/model/device_list.rb (hunk 4 above)
Comment (tseglevskiy) [reply lib/stf/model/device_list.rb]: because it returns list, not boolean. became `select_not_healthy`

File: lib/stf/model/device.rb (hunk 6 above)
Comment (tseglevskiy) [reply lib/stf/model/device.rb]: sure

File: lib/stf/model/device_list.rb (hunk 2 above)
Comment (tseglevskiy) [reply lib/stf/model/device_list.rb]: sure

**Pull request merged**